Implement lightning related stuff

Postremus opened the pull request:
Implement LightningStrike.Spigot

Make entities transmute when hit by lightning
pig -> pigzombie
villager -> witch
creeper -> charged creeper

related to #504

Postremus committed:
Implement LightningStrike.Spigot

Make Entities transmutate when hit by lightning
pig -> pigzombie
villager -> witch
creeper -> charged creeper

related to #504

Postremus committed:
Merge branch 'master' into #504-lightning

Postremus committed:
remove unused import

aramperes commented on src/main/java/net/glowstone/entity/EntityRegistry.java:
```diff
@@ -52,8 +52,8 @@
                     .put(ItemFrame.class, GlowItemFrame.class)
-                    .put(LeashHitch.class, GlowLeashHitch.class)
-                    //TODO: Lightning
+                    .put(LeashHitch.class, GlowLeashHitch.class)                     
```
Remove extra whitespace at end of line 55

aramperes commented on src/main/java/net/glowstone/entity/GlowLightningStrike.java:
```diff
@@ -39,10 +40,25 @@
+     * Whether the lightning strike is silent
+     */
+    private boolean isSilent;
+    private LightningStrike.Spigot spigot = new LightningStrike.Spigot() {
```
I think this can be final.

aramperes commented on src/main/java/net/glowstone/entity/passive/GlowVillager.java:
```diff
@@ -99,4 +102,14 @@ protected Sound getDeathSound() {
+    @Override
+    public void damage(double amount, Entity source, DamageCause cause) {
+        if (!DamageCause.LIGHTNING.equals(cause)) {
+            super.damage(amount, source, cause);
```
This would call the EntityDamageEvent which can be cancelled by plugins -- we should probably have an internal method that returns false when that event is cancelled, so subsequent actions (like converting) can be cancelled too.

Postremus replied:
resolved this in 47cf385

Postremus committed:
ZombiePig do take lightning damage
but no fire / lava damage
witches take damage upon spawning
Don't try to damage already dead entities
